lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -371,6 +371,12 @@ export class UserAgentApplication {
     // Track login in progress
     this.loginInProgress = true;
 
+    // request can be optional for login calls, hence check for the same
+    let reqState: string;
+    if (request) {
```

Comment:
Should the function parameter be declared as
`request?: AuthenticationParameters`
instead to indicate that it is valid for it to be undefined?

Reply:
@AE-MS Thanks for reviewing! It is true that making the request parameter optional would allow us to pass it as undefined, which is what we have exposed in the loginRedirect function. However, that doesn't solve the issue of trying to retrieve state from a possibly null request object, which is what the code here does. We will try to find a more elegant solution for this when we perform some refactoring in the future.

Reply:
@AE-MS Updated the login helper functions too to have optional parameters, to clear any confusion extending what @pkanher617 mentioned - "request" is optional only for login() functions but not for acquireToken() calls.